torch/_inductor/comm_analysis.py

```diff
@@ -69,18 +71,29 @@ def get_collective_type(node: ir.IRNode) -> NCCL_COLL:
     return get_collective_type_from_kernel_name(name)
 
 
-def get_size_numel(size: torch.Size, fallback: int = 4096 * 4096) -> int:
+def get_ir_node_size_numel(size: torch.Size, fallback: int = 4096 * 4096) -> int:
     numel = sympy_product(size)
     if isinstance(numel, sympy.Integer):
         return int(numel)
-
     return V.graph.sizevars.size_hint(numel, fallback=fallback)
 
 
+def get_fx_node_size_numel(
+    size: torch.Size, shape_env:  Optional[ShapeEnv] = None, fallback: int = 4096 * 4096
```

Comment:
mm do not need to pass shape_env there is a a global size_hint that recevies that we can also add falllback to it if needed . 
wonder why hint_int did not work, maybe due to the wrong sympy_product?